dotnet/src/SemanticKernel/TemplateEngine/PromptTemplateEngine.cs

```diff
@@ -105,20 +119,32 @@ public IList<Block> RenderVariables(IList<Block> blocks, ContextVariables? varia
         SKContext executionContext)
     {
         this._log.LogTrace("Rendering code");
-        var updatedBlocks = new List<Block>();
-        foreach (var block in blocks)
+
+        var result = new List<Block?>();
+        List<Task<string>> tasks = new();
+        List<int> indexes = new();
+
+        for (var i = 0; i < blocks.Count; i++)
         {
+            var block = blocks[i];
             if (block.Type != BlockTypes.Code)
             {
-                updatedBlocks.Add(block);
+                result.Add(block);
             }
             else
             {
-                var codeResult = await ((ICodeRendering)block).RenderCodeAsync(executionContext).ConfigureAwait(false);
-                updatedBlocks.Add(new TextBlock(codeResult, this._log));
+                result.Add(null);
```

Comment:
nit: Instead of adding `null` to the list we can define `result` as array `new Block[blocks.Count]`. In this case array items will be `null` by default and we will set result blocks directly by index.